packages/react-components/src/components/TextArea/TextArea.css

```diff
@@ -0,0 +1,130 @@
+  color: var(--typography-color-secondary);
+  display: grid;
+  gap: var(--layout-margin-medium);
+}
+
+/* Description label */
+.bcds-react-aria-TextArea--Description .text {
+  justify-self: flex-start;
+  grid-row: 1;
+}
+
+/* Character counter */
+.bcds-react-aria-TextArea--Description .counter {
+  justify-self: flex-end;
+  grid-row: 1;
+}
+
+/* Styles for the input field container */
+.bcds-react-aria-TextArea--Container {
+  color: var(--typography-color-primary);
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: var(--layout-margin-small);
+  background: var(--surface-color-forms-default);
+  box-sizing: border-box;
+  border: var(--layout-border-width-small) solid
+    var(--surface-color-border-default);
+  border-radius: var(--layout-border-radius-medium);
+  margin: var(--layout-margin-small) var(--layout-margin-none);
+  padding: var(--layout-padding-small) 12px;
+}
+
+/* Text input field */
+.bcds-react-aria-TextArea--Input {
+  font: var(--typography-regular-body);
+  padding: var(--layout-padding-none);
+  color: var(--typography-color-primary);
+  border: none;
+  flex-grow: 1;
+}
+
+.bcds-rect-aria-TextArea--Input::placeholder {
+  color: var(--typography-color-placeholder);
+}
+
+/* Hover and focus states */
+.bcds-react-aria-TextArea--Container
+  > .bcds-react-aria-TextArea--Input[data-focused] {
+  outline: none;
+}
+
+.bcds-react-aria-TextArea--Container:focus-within {
+  border-radius: var(--layout-border-radius-large);
+  border: var(--layout-border-width-small) solid
+    var(--surface-color-border-active);
+  outline: solid var(--layout-border-width-medium)
+    var(--surface-color-border-active);
+  outline-offset: var(--layout-margin-hair);
+}
+
+.bcds-react-aria-TextArea--Container:hover {
+  border-color: var(--surface-color-border-dark);
+}
+
+/* Disabled and invalid states */
+.bcds-react-aria-TextArea[data-disabled]
+  > .bcds-react-aria-TextArea--Container {
+  background: var(--surface-color-forms-disabled);
+  cursor: not-allowed;
+}
+
+.bcds-react-aria-TextArea--Input[data-disabled] {
+  cursor: not-allowed;
+}
+
+.bcds-react-aria-TextArea[data-invalid] > .bcds-react-aria-TextArea--Container {
+  border-radius: var(--layout-border-radius-medium);
+  border: 1px solid var(--support-border-color-danger);
+  background: var(--surface-color-forms-default);
+}
+
+.bcds-react-aria-TextArea[data-readonly]
+  > .bcds-react-aria-TextArea--Container {
+  background: var(--surface-color-forms-disabled);
+}
+
+.bcds-react-aria-TextArea[data-readonly]
+  > .bcds-react-aria-TextArea--Container
+  > .bcds-react-aria-TextArea--Input {
+  background: var(--surface-color-forms-disabled);
+}
+
+/* Styles for the error message slot */
+.bcds-react-aria-TextArea--Error {
+  font: var(--typography-regular-small-body);
+  color: var(--typography-color-danger);
+}
+
+/* Warning icon */
+.bcds-react-aria-TextArea--Error::before {
+  content: url("../src/assets/icon-error.svg");
```

Comment:
Unfortunately this won't work as written when we ship this component because that `url()` SVG file referenced here won't exist in the code that's outputted by our `rollup` script. That line makes it way into the code, but there will be no such target SVG file at run time, so this will silently fail (because it's in CSS and not a JS import that's failing):

<img width="1840" alt="Error icon before reference in dist code" src="https://github.com/user-attachments/assets/5446e756-5f3e-4c43-b82c-c2ccb2b44ac8">

To solve, this is what I would do:

1. Remove `icon-error.svg` and instead add `src/components/SvgExclamation` similar to how we're shipping the `SvgBcLogo` now, as a React component that returns an `<svg>` tag.
    - Avoid the name `error` for this icon because it's potentially useful as a generic exclamation icon until it has other "error context" (red color, error message next to it, etc).
    - We want to avoid hard-coding `id` values in the SVG JSX if possible - see `SvgBcLogo` as an example. Most likely we only need an `id`on the `<svg>` parent tag, as a target for future automated tests.
    - To let people style the icon after the fact, we want to allow coloring the icon with CSS outside of the context of the SVG JSX file, so we should set `fill` to `currentColor`.
    - If you want to keep `icon-error.svg` in the assets folder as a reference to what `SvgExclamation` came from, we can do that like we've done with the other SVGs in that directory, but we should still make the `id` and `currentColor` changes in that copy of the file.
2. In the `TextArea` component file, pass the render prop `isInvalid` and use that to conditionally render a fragment `<></>` (or `<div>`, or whatever) around the `<SvgExclamation>` and the `<FieldError>{errorMessage}</FieldError>`. 
   - This should solve the issue you were having where `<FieldError>` can't have other `children`:
     <img width="967" alt="FieldError TypeScript error" src="https://github.com/user-attachments/assets/3a76f703-b850-46f7-9d0a-53d03db8f5f2">

Reply:
So, (1) is now done (see [1423f3e](https://github.com/bcgov/design-system/pull/400/commits/1423f3e45ce2f59b8cab83461543acc4d739e493), [95c8ac3](https://github.com/bcgov/design-system/pull/400/commits/95c8ac3d7cfed18eb8e58cf5970051d16dec9434) and [05f9a25](https://github.com/bcgov/design-system/pull/400/commits/05f9a253f237077bafb53b99cd6460a049f2c273)), and I made a quick attempt at (2), but seemed to immediately hit up against the same issue I was having before.

There seems to be an related/underlying issue here that if the `<FieldError>` is anything other than the direct child of its parent component (in this case, the `<TextField>`), it just stops working entirely. So when it's wrapped in a fragment or a div, the error message won't render at all. I encountered this when I was first trying to debug this, and it's what led me towards the SVG/CSS approach.

[RAC documentation](https://react-spectrum.adobe.com/react-aria/TextField.html#validation) is a little oblique on this, only says:

> To display validation errors, add a <FieldError> element as a child of the TextField.

Reply:
Implemented in [d072d78](https://github.com/bcgov/design-system/pull/400/commits/d072d78dd24718c3ad214a2f7d91c13c4398d1fb)